packages/java-parser/src/productions/expressions.js

```diff
@@ -305,14 +309,18 @@ function defineRules($, t) {
       { ALT: () => $.CONSUME(t.Super) }
     ]);
 
-    let isRefTypeInMethodRef = false;
-    // Performance optimization, only perform this backtracking when a '<' is found
-    // TODO: performance optimization evaluation: avoid doing this backtracking for every "<" encountered.
-    //       we could do it once (using global state) per "fqnOrRefType"
-    // We could do it only once for
-    if (tokenMatcher($.LA(1).tokenType, t.Less)) {
-      isRefTypeInMethodRef = this.BACKTRACK_LOOKAHEAD($.isRefTypeInMethodRef);
-    }
+    const isRefTypeInMethodRef = $.ACTION(() => {
+      let isRefTypeInMethodRef = false;
+      // Performance optimization, only perform this backtracking when a '<' is found
+      // TODO: performance optimization evaluation: avoid doing this backtracking for every "<" encountered.
+      //       we could do it once (using global state) per "fqnOrRefType"
+      // We could do it only once for
+      if (tokenMatcher($.LA(1).tokenType, t.Less)) {
+        isRefTypeInMethodRef = this.BACKTRACK_LOOKAHEAD($.isRefTypeInMethodRef);
```

Comment:
same as above maybe too much is wrapped in $.ACTION
may have small runtime performance impact.